src/core/annotation.js

```diff
@@ -2245,6 +2250,10 @@ class HighlightAnnotation extends MarkupAnnotation {
           },
         });
       }
+    } else {
+      // QuadPoints is required
+      this._isInvalid = true;
```

Comment:
This name feels very "generic", while it's however only used together with QuadPoints.

Couldn't we simply set `this.data.quadPoints = null;` explicitly here, and then add a `this.data.quadPoints !== null` check to the `_isViewable`/`_isPrintable` helpers instead?

Reply:
So for any reason, I can't attach a file here so I filed an issue #12506.